rig-neo4j/src/lib.rs

```diff
@@ -0,0 +1,272 @@
+                serde_json::Value::Number(num) => {
+                    if let Some(i) = num.as_i64() {
+                        BoltType::Integer(BoltInteger::new(i))
+                    } else if let Some(f) = num.as_f64() {
+                        BoltType::Float(BoltFloat::new(f))
+                    } else {
+                        println!("Couldn't map to BoltType, will ignore.");
+                        BoltType::Null(BoltNull) // Handle unexpected number type
+                    }
+                }
+                serde_json::Value::String(s) => BoltType::String(BoltString::new(&s)),
+                serde_json::Value::Array(arr) => BoltType::List(
+                    arr.iter()
+                        .map(|v| v.to_bolt_type())
+                        .collect::<Vec<BoltType>>()
+                        .into(),
+                ),
+                serde_json::Value::Object(obj) => {
+                    let mut bolt_map = BoltMap::new();
+                    for (k, v) in obj {
+                        bolt_map.put(BoltString::new(&k), v.to_bolt_type());
+                    }
+                    BoltType::Map(bolt_map)
+                }
+            },
+            Err(_) => {
+                println!("Couldn't serialize to JSON, will ignore.");
+                BoltType::Null(BoltNull) // Handle serialization error
+            }
+        }
+    }
+}
+
+impl Neo4jClient {
+    pub fn new(graph: Graph) -> Self {
+        Self { graph }
+    }
+
+    pub async fn connect(uri: &str, user: &str, password: &str) -> Result<Self, VectorStoreError> {
+        let graph = Graph::new(uri, user, password)
+            .await
+            .map_err(|e| VectorStoreError::DatastoreError(Box::new(e)))?;
+        Ok(Self { graph })
+    }
+
+    pub async fn from_config(config: Config) -> Result<Self, VectorStoreError> {
+        let graph = Graph::connect(config)
+            .await
+            .map_err(|e| VectorStoreError::DatastoreError(Box::new(e)))?;
+        Ok(Self { graph })
+    }
+
+    /// Returns a `Neo4jVectorIndex` that mirrors an existing Neo4j Vector Index.
+    ///
+    /// An index (of type "vector") of the same name as `index_name` must already exist for the Neo4j database.
+    /// See the Neo4j [documentation (Create vector index)](https://neo4j.com/docs/genai/tutorials/embeddings-vector-indexes/setup/vector-index/) for more information on creating indexes.
+    ///
+    /// ❗IMPORTANT: The index must be created with the same embedding model that will be used to query the index.
+    pub fn index<M: EmbeddingModel>(
+        &self,
+        model: M,
+        index_config: IndexConfig,
+        search_params: SearchParams,
+    ) -> Neo4jVectorIndex<M> {
+        Neo4jVectorIndex::new(self.graph.clone(), model, index_config, search_params)
+    }
+}
+
+#[allow(dead_code)]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use display;
+    use display::SearchResult;
```

Comment:
The only test that needs this... is a test for the display functionality! So remove both the test and this import

Reply:
@mateobelanger the `display` import is still present in the `test` module